File: src/main/audio/sounds.ts
import { execFile } from 'child_process';

/**
 * Plays a macOS system sound. Non-blocking, fire-and-forget.
 * Sound names: Tink, Pop, Basso, Bottle, Frog, Funk, Glass, Hero, Morse, Ping, Purr, Sosumi, Submarine
 */
function playSystemSound(name: string): void {
  const soundPath = `/System/Library/Sounds/${name}.aiff`;
  execFile('afplay', [soundPath], (err) => {
    if (err) console.warn(`[sounds] Failed to play ${name}:`, err.message);
  });
}

export function playRecordingStart(): void {
  playSystemSound('Tink');
}

export function playRecordingStop(): void {
  playSystemSound('Pop');
}

export function playError(): void {
  playSystemSound('Basso');
}

export function playSuccess(): void {
  playSystemSound('Glass');
}

File: src-tauri/src/audio/sounds.rs
use std::process::Command;

fn play_system_sound(name: &str) {
    let path = format!("/System/Library/Sounds/{}.aiff", name);
    let _ = Command::new("afplay").arg(&path).spawn();
}

pub fn play_recording_start() {
    play_system_sound("Tink");
}

pub fn play_recording_stop() {
    play_system_sound("Pop");
}

pub fn play_error() {
    play_system_sound("Basso");
}

pub fn play_success() {
    play_system_sound("Glass");
}
